core/trino-main/src/test/java/io/trino/sql/query/TestColumnMask.java

```diff
@@ -218,6 +218,96 @@ public void testMultipleMasksOnDifferentColumns()
                 .matches("VALUES (BIGINT '-370', 'X')");
     }
 
+    @Test
+    public void testMultipleMasksUsingOtherMaskedColumns()
+    {
+        String query = "SELECT comment, orderstatus, clerk FROM orders WHERE orderkey = 1";
+        String expected = "VALUES (CAST('nstructions sleep furiously among ' as varchar(79)), 'O', 'Clerk#000000951')";
+
+        // mask "clerk" and "orderstatus" using "comment" ("comment" appears after both in table definition)
+        accessControl.reset();
+        accessControl.columnMask(
+                new QualifiedObjectName(CATALOG, "tiny", "orders"),
+                "comment",
+                USER,
+                new ViewExpression(USER, Optional.empty(), Optional.empty(), "cast(regexp_replace(comment,'(password: [^ ]+)','password: ****') as varchar(79))"));
```

Comment:
I don't quite get the purpose of this masking.
The column `comment` has the value `nstructions sleep furiously among `.
Why do you check for `passwod` in this text?

Reply:
I want to mask some values in `comment` column (in this case the `password` value) and also to use `comment` column as a reference for the decision about masking other columns (we called it "conditional masking")

In a real example there is a json `payload` column with some attributes that should be masked and a `country`  attribute that used as a reference for masking other columns.

Reply:
> The column comment has the value `nstructions sleep furiously among `.

This value doesn't contain a `password`  substring. So, I guess the masking is pointless in this case. Or am I missing something?

Reply:
I found that issue happens in [InlineProjections#inlineProjections](https://github.com/trinodb/trino/blob/f01a6fa2ac7e611e21732479f546ff5fd564eb32/core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/InlineProjections.java#L117-L125)
Maybe it relates on how column masks are [added](https://github.com/trinodb/trino/blob/f01a6fa2ac7e611e21732479f546ff5fd564eb32/core/trino-main/src/main/java/io/trino/sql/planner/RelationPlanner.java#L299-L317)

Reply:
> > The column comment has the value `nstructions sleep furiously among `.
> 
> This value doesn't contain a `password` substring. So, I guess the masking is pointless in this case. Or am I missing something?

You are right. That's why I added the last test (L287:L309)

Reply:
I had a look again on the test and I am sorry to say that I still don't quite understand why isn't there any masking in the result of the queries.
Shouldn't the test demonstrate that the `*` is masking the content of the returned columns on which the masking should apply?

Reply:
> Shouldn't the test demonstrate that the `*` is masking the content of the returned columns on which the masking should apply?

You are right. I fixed it now. Take a look on the tests with `orderkey = 39` (L289:L332)

Reply:
> > The column comment has the value `nstructions sleep furiously among `.
> 
> This value doesn't contain a `password` substring. So, I guess the masking is pointless in this case. Or am I missing something?

That's right. masking is pointless for this specific row, but I added it to demonstrate the issue when adding the masking on `orderstatus` while refering `clerk` (L287)

Reply:
I see now. 

The purpose of the test is currently just to demonstrate that the masking doesn't work properly.  Thank you for the patience in trying to explain the purpose of the tests.